src/backend/mpi/swdsm.cpp

```diff
@@ -1228,20 +1226,20 @@ void argo_reset_stats(){
 	stats.ssd_time = 0;
 
 	// Clear the cache lock statistics
-	for( auto& cache_lock : cache_locks ){
+	for(auto& cache_lock : cache_locks){
 		cache_lock.reset_stats();
 	}
 
 	// Clear the sharer lock statistics
-	for( std::size_t i = 0; i < mpi_windows; i++ ) {
-		for( int j = 0; j < numtasks; j++ ) {
+	for(std::size_t i = 0; i < mpi_windows; i++) {
+		for(argo::node_id_t j = 0; j < numtasks; j++) {
 			mpi_lock_sharer[i][j].reset_stats();
 		}
 	}
 
 	// Clear the data lock statistics
-	for( std::size_t i = 0; i < mpi_windows; i++ ) {
-		for( int j = 0; j < numtasks; j++ ) {
+	for(std::size_t i = 0; i < mpi_windows; i++) {
+		for(argo::node_id_t j = 0; j < numtasks; j++) {
```

Comment:
Are `numtasks` of `node_id_t` type?

Reply:
It is an `int`, so apparently it is not (although `node_id_t` happens to be an int too).

Since we regularly compare with `< numtasks` in for loops, we can't mix signed and unsigned. For that reason it would make sense if `numtasks` was also in the form of `node_id_t`, or if , even though a number of nodes is technically not a node id. Open for other suggestions such as changing all loop indexing comparing with `numtasks` to `int`.

Reply:
Resolved by introducing a `num_nodes_t` type to use for `numtasks`.